### PR TITLE
fix(Omnichannel): Omnichannel agent `take it` button

### PR DESCRIPTION
**Pull request by tiagoevanp**
## Proposed changes (including videos or screenshots)

"Take it" button should be disabled when agent status is set to offline

image

## Further comments

[CORE-256](https://rocketchat.atlassian.net/browse/CORE-256)

[CORE-256]: https://rocketchat.atlassian.net/browse/CORE-256?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/meteor/tests/e2e/omnichannel/omnichannel-take-chat.spec.ts [hunk 1]
```diff
@@ -25,50 +27,66 @@ test.describe('omnichannel-takeChat', () => {
...
+		await poLiveChat.closeChat();
+	});
+
+	test('should user not receive a new chat when offline status and livechat disabled when agent idle', async ({ api }) => {
+		await api.post('/settings/Livechat_enabled_when_agent_idle', { value: false }).then((res) => expect(res.status()).toBe(200));
+		await agent.poHomeChannel.sidenav.switchStatus('offline');
+
+		await poLiveChat.openChatAndSendMessage(newVisitor, 'this_a_test_message_from_user');
+
+		await sleep(2000);
```
Comment (MarcosSpessatto) [inline apps/meteor/tests/e2e/omnichannel/omnichannel-take-chat.spec.ts:71]: Would not this be a problem? 🤔

Comment (KevLehman) [inline apps/meteor/tests/e2e/omnichannel/omnichannel-take-chat.spec.ts:71]: Since u marked the user as offline, there should be no way for the user to have items on his queued chats section, so the sleep shouldn't be needed.

File: apps/meteor/tests/e2e/omnichannel/omnichannel-take-chat.spec.ts [hunk 2]
```diff
@@ -16,7 +18,7 @@ test.describe('omnichannel-takeChat', () => {
 		await Promise.all([
 			await api.post('/livechat/users/agent', { username: 'user1' }).then((res) => expect(res.status()).toBe(200)),
 			await api.post('/settings/Livechat_Routing_Method', { value: 'Manual_Selection' }).then((res) => expect(res.status()).toBe(200)),
-			await api.post('/settings/Livechat_enabled_when_agent_idle', { value: false }).then((res) => expect(res.status()).toBe(200)),
+			await api.post('/settings/Livechat_enabled_when_agent_idle', { value: true }).then((res) => expect(res.status()).toBe(200)),
```
Comment (KevLehman) [inline apps/meteor/tests/e2e/omnichannel/omnichannel-take-chat.spec.ts:21]: Restore this one to the previous value after your tests.

File: apps/meteor/tests/e2e/omnichannel/omnichannel-take-chat.spec.ts [hunk 3]
```diff
@@ -25,50 +27,66 @@ test.describe('omnichannel-takeChat', () => {
...
+
 		await page.goto('/livechat');
-		await poLiveChat.openLiveChat();
-		await poLiveChat.sendMessage(newVisitor, false);
-		await poLiveChat.onlineAgentMessage.type('this_a_test_message_from_user');
-		await poLiveChat.btnSendMessageToOnlineAgent.click();
 	});
 
-	test('expect "user1" to be able to take the chat from the queue', async () => {
+	test('should user take the chat', async () => {
```
Comment (KevLehman) [inline apps/meteor/tests/e2e/omnichannel/omnichannel-take-chat.spec.ts:51]:
```suggestion
	test('user should take the chat', async () => {
```

File: apps/meteor/tests/e2e/omnichannel/omnichannel-take-chat.spec.ts [hunk 4]
```diff
@@ -25,50 +27,66 @@ test.describe('omnichannel-takeChat', () => {
...
 		await agent.poHomeChannel.content.btnTakeChat.click();
 
 		await agent.poHomeChannel.sidenav.openChat(newVisitor.name);
 		await expect(agent.poHomeChannel.content.btnTakeChat).not.toBeVisible();
 		await expect(agent.poHomeChannel.content.inputMessage).toBeVisible();
+
+		await poLiveChat.closeChat();
+	});
+
+	test('should user not receive a new chat when offline status and livechat disabled when agent idle', async ({ api }) => {
```
Comment (KevLehman) [inline apps/meteor/tests/e2e/omnichannel/omnichannel-take-chat.spec.ts:65]:
```suggestion
	test('user should not receive a new chat when offline status and livechat disabled when agent idle', async ({ api }) => {
```

File: apps/meteor/tests/e2e/omnichannel/omnichannel-take-chat.spec.ts [hunk 5]
```diff
@@ -25,50 +27,66 @@ test.describe('omnichannel-takeChat', () => {
...
+		await poLiveChat.openChatAndSendMessage(newVisitor, 'this_a_test_message_from_user');
+
+		await sleep(2000);
+
+		await expect(agent.poHomeChannel.sidenav.getQueuedChat(newVisitor.name)).toHaveCount(0);
 	});
 
-	test('expect "user1" to not able able to take chat from queue in-case its user status is offline', async () => {
-		// make "user-1" offline
+	test('should user not take the chat when offline status', async ({ api }) => {
```
Comment (KevLehman) [inline apps/meteor/tests/e2e/omnichannel/omnichannel-take-chat.spec.ts:76]:
```suggestion
	test('user should not take the chat when offline status', async ({ api }) => {
```

**Review by KevLehman — CHANGES_REQUESTED** [review]

File: apps/meteor/tests/e2e/omnichannel/omnichannel-take-chat.spec.ts (hunk 2 above)
Comment (tiagoevanp) [reply apps/meteor/tests/e2e/omnichannel/omnichannel-take-chat.spec.ts:21]: The default value is true on this 🤔

File: apps/meteor/tests/e2e/omnichannel/omnichannel-take-chat.spec.ts (hunk 1 above)
Comment (tiagoevanp) [reply apps/meteor/tests/e2e/omnichannel/omnichannel-take-chat.spec.ts:71]: I'm trying to avoid an always falsy test... Since the message from livechat could take some time to be received by RC instance 🤔

Comment (KevLehman) [reply apps/meteor/tests/e2e/omnichannel/omnichannel-take-chat.spec.ts:71]: Instead I think you can check that when you go offline the whole "Queued chats" section to dissapear
There you'll be sure a new chat won't be there.

File: apps/meteor/tests/e2e/omnichannel/omnichannel-take-chat.spec.ts (hunk 2 above)
Comment (tiagoevanp) [reply apps/meteor/tests/e2e/omnichannel/omnichannel-take-chat.spec.ts:21]: This is to ensure that the status is correct before starting the test.

File: apps/meteor/tests/e2e/omnichannel/omnichannel-take-chat.spec.ts (hunk 1 above)
Comment (tiagoevanp) [reply apps/meteor/tests/e2e/omnichannel/omnichannel-take-chat.spec.ts:71]: this is the user status, not livechat status... so the queued chats will not disappear...

Comment (KevLehman) [reply apps/meteor/tests/e2e/omnichannel/omnichannel-take-chat.spec.ts:71]: Arg, thought that switch was for the other status 💢 

Ok 🤔 think there's no other way :(

File: apps/meteor/tests/e2e/omnichannel/omnichannel-take-chat.spec.ts [hunk 6]
```diff
@@ -1,12 +1,14 @@
...
 
+import { sleep } from '../../data/livechat/utils';
 import { createAuxContext } from '../fixtures/createAuxContext';
 import { Users } from '../fixtures/userStates';
 import { OmnichannelLiveChat, HomeOmnichannel } from '../page-objects';
 import { test, expect } from '../utils/test';
 
-test.describe('omnichannel-takeChat', () => {
+
+test.describe('omnichannel-take-chat', () => {
```
Comment (ggazzo) [inline apps/meteor/tests/e2e/omnichannel/omnichannel-take-chat.spec.ts:11]: stay with the same name, otherwise this is going to be duplicated

File: apps/meteor/tests/e2e/omnichannel/omnichannel-take-chat.spec.ts (hunk 1 above)
Comment (ggazzo) [reply apps/meteor/tests/e2e/omnichannel/omnichannel-take-chat.spec.ts:71]: playwright already waits for the elements, you can modify the timeout time if you believe it will make a difference, but honestly, waiting longer should not be the solution that will save your test

File: apps/meteor/tests/e2e/omnichannel/omnichannel-take-chat.spec.ts [hunk 7]
```diff
@@ -25,50 +27,66 @@ test.describe('omnichannel-takeChat', () => {
...
+		await poLiveChat.openChatAndSendMessage(newVisitor, 'this_a_test_message_from_user');
+
 		await agent.poHomeChannel.sidenav.switchStatus('offline');
 
 		await agent.poHomeChannel.sidenav.openQueuedOmnichannelChat(newVisitor.name);
+
 		await expect(agent.poHomeChannel.content.btnTakeChat).toBeVisible();
+
 		await agent.poHomeChannel.content.btnTakeChat.click();
+		await expect(agent.page.locator('role=alert')).toBeVisible();
```
Comment (ggazzo) [inline apps/meteor/tests/e2e/omnichannel/omnichannel-take-chat.spec.ts:88]: to broad selector it could be any alert in the page, it can introduce new flaky test in the future if we have 2+ in the page

File: apps/meteor/tests/e2e/page-objects/fragments/home-sidenav.ts [hunk 8]
```diff
@@ -119,7 +123,7 @@ export class HomeSidenav {
 
 	// Note: this is a workaround for now since queued omnichannel chats are not searchable yet so we can't use openChat() :(
 	async openQueuedOmnichannelChat(name: string): Promise<void> {
-		await this.page.locator('[data-qa="sidebar-item-title"]', { hasText: name }).first().click();
+		await this.getQueuedChat(name).click();
```
Comment (ggazzo) [inline apps/meteor/tests/e2e/page-objects/fragments/home-sidenav.ts:126]: Looking at it like this, it looks like we have 2 equal selectors?

File: apps/meteor/tests/e2e/page-objects/omnichannel-livechat.ts [hunk 9]
```diff
@@ -142,6 +142,13 @@ export class OmnichannelLiveChat {
 		return this.page.getByRole('link', { name: fileName });
 	}
 
+	public async openChatAndSendMessage(liveChatUser: { name: string; email: string }, message = '') {
+		await this.openLiveChat();
+		await this.sendMessage(liveChatUser, false);
+		await this.onlineAgentMessage.fill(message);
+		await this.btnSendMessageToOnlineAgent.click();
```
Comment (ggazzo) [inline apps/meteor/tests/e2e/page-objects/omnichannel-livechat.ts:149]: I really don't like multi-level pageobjects, it diverts the focus of the test, it forces the tester to understand the test programming

**Review by ggazzo — CHANGES_REQUESTED** [review]

**Review by KevLehman — APPROVED** [review]

**Review by MarcosSpessatto — CHANGES_REQUESTED** [review]
Can we add a changeset?

File: apps/meteor/client/views/room/composer/ComposerOmnichannel/ComposerOmnichannelInquiry.tsx [hunk 10]
```diff
@@ -36,7 +36,7 @@ export const ComposerOmnichannelInquiry = (): ReactElement => {
 	return (
 		<MessageFooterCallout aria-busy={result.isLoading}>
 			<MessageFooterCalloutContent>{t('you_are_in_preview_mode_of_incoming_livechat')}</MessageFooterCalloutContent>
-			<MessageFooterCalloutAction disabled={result.isLoading} onClick={handleTakeInquiry}>
+			<MessageFooterCalloutAction disabled={result.isLoading || user?.status === 'offline'} onClick={handleTakeInquiry}>
```
[the remark this answers is not in the document]
Comment (tiagoevanp) [reply apps/meteor/client/views/room/composer/ComposerOmnichannel/ComposerOmnichannelInquiry.tsx:39]: Well, the result from the endpoint should be an error, prevent call is obviously a better decision than make it to receive an error. I don't know what mean by `service status` 🤔

File: apps/meteor/tests/e2e/page-objects/fragments/home-sidenav.ts [hunk 11]
```diff
@@ -129,11 +133,6 @@ export class HomeSidenav {
 		expect(newStatus).toBe(status === 'offline' ? StatusTitleMap.offline : StatusTitleMap.online);
 	}
 
-	// Note: this is a workaround for now since queued omnichannel chats are not searchable yet so we can't use openChat() :(
-	async openQueuedOmnichannelChat(name: string): Promise<void> {
```
[the remark this answers is not in the document]
Comment (tiagoevanp) [reply apps/meteor/tests/e2e/page-objects/fragments/home-sidenav.ts:133]: I think is better to keep locators inside of the fragments and, at least, the majority of logic inside of the test itself...

File: apps/meteor/client/views/room/composer/ComposerOmnichannel/ComposerOmnichannelInquiry.tsx (hunk 10 above)
[the remark this answers is not in the document]
Comment (KevLehman) [reply apps/meteor/client/views/room/composer/ComposerOmnichannel/ComposerOmnichannelInquiry.tsx:39]: Status = user Status
Service Status = Status livechat


Yeah, but its always good to confirm :) most likely would be a yes but better be sure

File: apps/meteor/tests/e2e/page-objects/fragments/home-sidenav.ts (hunk 11 above)
[the remark this answers is not in the document]
Comment (KevLehman) [reply apps/meteor/tests/e2e/page-objects/fragments/home-sidenav.ts:133]: Then we should keep the comment because the logic remains the same

[the remark this answers is not in the document]
Comment (tiagoevanp) [reply apps/meteor/tests/e2e/page-objects/fragments/home-sidenav.ts:133]: This comment adds nothing to the test itself, and this is not a workaround since we will not have search behavior for queued chats 🤔

File: apps/meteor/client/views/room/composer/ComposerOmnichannel/ComposerOmnichannelInquiry.tsx (hunk 10 above)
[the remark this answers is not in the document]
Comment (tiagoevanp) [reply apps/meteor/client/views/room/composer/ComposerOmnichannel/ComposerOmnichannelInquiry.tsx:39]: If I'm not wrong... when you turn off the livechat status, queued chats desapear from sidebar 🤔 So it's unnecessary to test in `take button` test context

[the remark this answers is not in the document]
Comment (KevLehman) [reply apps/meteor/client/views/room/composer/ComposerOmnichannel/ComposerOmnichannelInquiry.tsx]: Yeah, when you turn off livechat, then you cannot see the queue. BUT, if you're seeing a chat, it won't dissapear from your screen, in that context, you can still "take" a chat.

Same happens for users who can see `current chats` page, they would be able to open a conversation and "take it".

We should also prevent that 😬 imho

[the remark this answers is not in the document]
Comment (tiagoevanp) [reply apps/meteor/client/views/room/composer/ComposerOmnichannel/ComposerOmnichannelInquiry.tsx]: updated!

File: apps/meteor/tests/e2e/page-objects/fragments/home-sidenav.ts (hunk 11 above)
[the remark this answers is not in the document]
Comment (tiagoevanp) [reply apps/meteor/tests/e2e/page-objects/fragments/home-sidenav.ts:133]: updated!

File: apps/meteor/client/views/room/composer/ComposerOmnichannel/ComposerOmnichannelInquiry.tsx [hunk 12]
```diff
@@ -38,7 +40,7 @@ export const ComposerOmnichannelInquiry = (): ReactElement => {
 			<MessageFooterCalloutContent>{t('you_are_in_preview_mode_of_incoming_livechat')}</MessageFooterCalloutContent>
 			<MessageFooterCalloutAction
 				{...(user?.status === 'offline' && { title: t('meteor_status_offline') })}
-				disabled={result.isLoading || user?.status === 'offline'}
+				disabled={result.isLoading || user?.status === 'offline' || !agentAvailable}
```
Comment (KevLehman) [inline apps/meteor/client/views/room/composer/ComposerOmnichannel/ComposerOmnichannelInquiry.tsx:43]: Pls add a tooltip to let the user know why the button is blocked

File: apps/meteor/client/views/room/composer/ComposerOmnichannel/ComposerOmnichannelInquiry.tsx [hunk 13]
```diff
@@ -36,7 +38,11 @@ export const ComposerOmnichannelInquiry = (): ReactElement => {
 	return (
 		<MessageFooterCallout aria-busy={result.isLoading}>
 			<MessageFooterCalloutContent>{t('you_are_in_preview_mode_of_incoming_livechat')}</MessageFooterCalloutContent>
-			<MessageFooterCalloutAction disabled={result.isLoading} onClick={handleTakeInquiry}>
+			<MessageFooterCalloutAction
+				{...((user?.status === 'offline' || !agentAvailable) && { title: t('meteor_status_offline') })}
```
Comment (KevLehman) [inline apps/meteor/client/views/room/composer/ComposerOmnichannel/ComposerOmnichannelInquiry.tsx:42]: Use different tooltips depending on what's the reason.

**Review by KevLehman — CHANGES_REQUESTED** [review]

File: packages/i18n/src/locales/en.i18n.json [hunk 14]
```diff
@@ -4958,7 +4958,7 @@
   "SMS": "SMS",
   "SMS_Description": "Enable and configure SMS gateways on your workspace.",
   "SMS_Default_Omnichannel_Department": "Omnichannel Department (Default)",
-  "SMS_Default_Omnichannel_Department_Description": "If set, all new incoming chats initiated by this integration will be routed to this department.  \nThis setting can be overwritten by passing department query param in the request.  \ne.g. `https://{{SERVER_URL}}/api/v1/livechat/sms-incoming/twilio?department={{Department Id or Name}}`.  \nNote: if you're using Department Name, then it should be URL safe.",
```
Comment (KevLehman) [inline packages/i18n/src/locales/en.i18n.json:4961]: Please remove the unrelated changes from these translation strings.

File: packages/i18n/src/locales/en.i18n.json [hunk 15]
```diff
@@ -6022,7 +6022,7 @@
   "Your_TOTP_has_been_reset": "Your Two Factor TOTP has been reset.",
   "Your_web_browser_blocked_Rocket_Chat_from_opening_tab": "Your web browser blocked Rocket.Chat from opening a new tab.",
   "Your_workspace_is_ready": "Your workspace is ready to use 🎉",
-  "Youre_not_a_part_of__channel__and_I_mentioned_you_there": "You're not a part of {{channel}} and I mentioned you there",
+  "Youre_not_a_part_of__channel__and_I_mentioned_you_there": "You'r not a part of {{channel}} and I mentioned you there",
```
Comment (tiagoevanp) [inline packages/i18n/src/locales/en.i18n.json:6025]:
```suggestion
  "Youre_not_a_part_of__channel__and_I_mentioned_you_there": "You're not a part of {{channel}} and I mentioned you there",
```

File: packages/i18n/src/locales/en.i18n.json [hunk 16]
```diff
@@ -4958,7 +4958,7 @@
   "SMS": "SMS",
   "SMS_Description": "Enable and configure SMS gateways on your workspace.",
   "SMS_Default_Omnichannel_Department": "Omnichannel Department (Default)",
-  "SMS_Default_Omnichannel_Department_Description": "If set, all new incoming chats initiated by this integration will be routed to this department.  \nThis setting can be overwritten by passing department query param in the request.  \ne.g. `https://{{SERVER_URL}}/api/v1/livechat/sms-incoming/twilio?department={{Department Id or Name}}`.  \nNote: if you're using Department Name, then it should be URL safe.",
+  "SMS_Default_Omnichannel_Department_Description": "If set, all new incoming chats initiated by this integration will be routed to this department.  \nThis setting can be overwritten by passing department query param in the request.  \ne.g. `https://{{SERVER_URL}}/api/v1/livechat/sms-incoming/twilio?department={{Department Id or Name}}`.  \nNote: if you'r using Department Name, then it should be URL safe.",
```
Comment (tiagoevanp) [inline packages/i18n/src/locales/en.i18n.json:4961]:
```suggestion
  "SMS_Default_Omnichannel_Department_Description": "If set, all new incoming chats initiated by this integration will be routed to this department.  \nThis setting can be overwritten by passing department query param in the request.  \ne.g. `https://{{SERVER_URL}}/api/v1/livechat/sms-incoming/twilio?department={{Department Id or Name}}`.  \nNote: if you're using Department Name, then it should be URL safe.",
```

File: packages/i18n/src/locales/en.i18n.json [hunk 17]
```diff
@@ -3080,7 +3080,7 @@
   "LDAP_Sync_User_Data_Channels_BaseDN": "LDAP Group BaseDN",
   "LDAP_Sync_User_Data_Channels_Description": "Enable this feature to automatically add users to a channel based on their LDAP group. If you would like to also remove users from a channel, see the option below about auto removing users.",
   "LDAP_Sync_User_Data_Channels_Enforce_AutoChannels": "Auto Remove Users from Channels",
-  "LDAP_Sync_User_Data_Channels_Enforce_AutoChannels_Description": "**Attention**: Enabling this will remove any users in a channel that do not have the corresponding LDAP group! Only enable this if you know what you're doing.",
+  "LDAP_Sync_User_Data_Channels_Enforce_AutoChannels_Description": "**Attention**: Enabling this will remove any users in a channel that do not have the corresponding LDAP group! Only enable this if you know what you'r doing.",
```
Comment (tiagoevanp) [inline packages/i18n/src/locales/en.i18n.json:3083]:
```suggestion
  "LDAP_Sync_User_Data_Channels_Enforce_AutoChannels_Description": "**Attention**: Enabling this will remove any users in a channel that do not have the corresponding LDAP group! Only enable this if you know what you're doing.",
```

File: packages/i18n/src/locales/en.i18n.json [hunk 18]
```diff
@@ -2300,9 +2300,9 @@
...
   "Federation_Matrix_error_applying_room_roles": "Something went wrong while applying the room roles over the federated network",
-  "Federation_Matrix_giving_same_permission_warning": "You're giving this user the same privileges as yourself, you will not be able to undo this change. Do you want to proceed?",
+  "Federation_Matrix_giving_same_permission_warning": "You'r giving this user the same privileges as yourself, you will not be able to undo this change. Do you want to proceed?",
   "Federation_Matrix_losing_privileges": "Losing privileges",
-  "Federation_Matrix_losing_privileges_warning": "You won't be able to undo this action, as you're demoting yourself. If you're the last privileged user you won't be able to regain this privilege. Do you want to proceed still?",
+  "Federation_Matrix_losing_privileges_warning": "You won't be able to undo this action, as you'r demoting yourself. If you'r the last privileged user you won't be able to regain this privilege. Do you want to proceed still?",
```
Comment (tiagoevanp) [inline packages/i18n/src/locales/en.i18n.json:2305]:
```suggestion
  "Federation_Matrix_losing_privileges_warning": "You won't be able to undo this action, as you're demoting yourself. If you'r the last privileged user you won't be able to regain this privilege. Do you want to proceed still?",
```

File: packages/i18n/src/locales/en.i18n.json [hunk 19]
```diff
@@ -2300,9 +2300,9 @@
   "Federation_Matrix_registration_file": "Registration File",
   "Federation_Matrix_registration_file_Alert": "Important: Enabling ephemeral events will make the server receive all the typing status of all users from all servers you are connected to.</br> To enable it, please update your registration file (.yaml file you are using to registrate Rocket.Chat to your home server), adding the following:</br> de.sorunome.msc2409.push_ephemeral: true",
   "Federation_Matrix_error_applying_room_roles": "Something went wrong while applying the room roles over the federated network",
-  "Federation_Matrix_giving_same_permission_warning": "You're giving this user the same privileges as yourself, you will not be able to undo this change. Do you want to proceed?",
+  "Federation_Matrix_giving_same_permission_warning": "You'r giving this user the same privileges as yourself, you will not be able to undo this change. Do you want to proceed?",
```
Comment (tiagoevanp) [inline packages/i18n/src/locales/en.i18n.json:2303]:
```suggestion
  "Federation_Matrix_giving_same_permission_warning": "You're giving this user the same privileges as yourself, you will not be able to undo this change. Do you want to proceed?",
```

File: packages/i18n/src/locales/en.i18n.json [hunk 20]
```diff
@@ -869,7 +869,7 @@
   "Call_again": "Call again",
   "Call_back": "Call back",
   "Call_not_found": "Call not found",
-  "Call_not_found_error": "This could happen when the call URL is not valid, or you're having connection issues. Please check with the source of the call URL and try again, or talk to your workspace administrator if the problem persists",
+  "Call_not_found_error": "This could happen when the call URL is not valid, or you'r having connection issues. Please check with the source of the call URL and try again, or talk to your workspace administrator if the problem persists",
```
Comment (tiagoevanp) [inline packages/i18n/src/locales/en.i18n.json:872]:
```suggestion
  "Call_not_found_error": "This could happen when the call URL is not valid, or you're having connection issues. Please check with the source of the call URL and try again, or talk to your workspace administrator if the problem persists",
```

File: packages/i18n/src/locales/en.i18n.json [hunk 21]
```diff
@@ -853,7 +853,7 @@
   "Business_Hours": "Business Hours",
   "Business_hours_enabled": "Business hours enabled",
   "Business_hours_is_disabled": "Business hours is disabled",
-  "Business_hours_is_disabled_description": "Enable business hours at the workspace admin panel to let customers know when you're available and when can they expect a response.",
+  "Business_hours_is_disabled_description": "Enable business hours at the workspace admin panel to let customers know when you'r available and when can they expect a response.",
```
Comment (tiagoevanp) [inline packages/i18n/src/locales/en.i18n.json:856]:
```suggestion
  "Business_hours_is_disabled_description": "Enable business hours at the workspace admin panel to let customers know when you're available and when can they expect a response.",
```

File: packages/i18n/src/locales/en.i18n.json [hunk 22]
```diff
@@ -2302,7 +2302,7 @@
   "Federation_Matrix_error_applying_room_roles": "Something went wrong while applying the room roles over the federated network",
   "Federation_Matrix_giving_same_permission_warning": "You're giving this user the same privileges as yourself, you will not be able to undo this change. Do you want to proceed?",
   "Federation_Matrix_losing_privileges": "Losing privileges",
-  "Federation_Matrix_losing_privileges_warning": "You won't be able to undo this action, as you're demoting yourself. If you're the last privileged user you won't be able to regain this privilege. Do you want to proceed still?",
+  "Federation_Matrix_losing_privileges_warning": "You won't be able to undo this action, as you're demoting yourself. If you'r the last privileged user you won't be able to regain this privilege. Do you want to proceed still?",
```
Comment (tiagoevanp) [inline packages/i18n/src/locales/en.i18n.json:2305]:
```suggestion
  "Federation_Matrix_losing_privileges_warning": "You won't be able to undo this action, as you're demoting yourself. If you're the last privileged user you won't be able to regain this privilege. Do you want to proceed still?",
```

File: packages/i18n/src/locales/en.i18n.json (hunk 14 above)
Comment (tiagoevanp) [reply packages/i18n/src/locales/en.i18n.json]: Done!

**Review by KevLehman — APPROVED** [review]

File: apps/meteor/client/views/room/composer/ComposerOmnichannel/ComposerOmnichannelInquiry.tsx [hunk 23]
```diff
@@ -32,11 +36,24 @@ export const ComposerOmnichannelInquiry = (): ReactElement => {
...
 	};
 
-	const t = useTranslation();
+	const title = useMemo(() => {
+		if (user?.status === 'offline') {
+			return t('meteor_status_offline');
+		}
+
+		if (!agentAvailable) {
+			return t('Youre_unavailable');
```
Comment (ggazzo) [inline apps/meteor/client/views/room/composer/ComposerOmnichannel/ComposerOmnichannelInquiry.tsx:45]: i think there is no need for two different reasons, not being available is valid for both cases

**Review by ggazzo — CHANGES_REQUESTED** [review]

File: apps/meteor/client/views/room/composer/ComposerOmnichannel/ComposerOmnichannelInquiry.tsx (hunk 23 above)
Comment (KevLehman) [reply apps/meteor/client/views/room/composer/ComposerOmnichannel/ComposerOmnichannelInquiry.tsx:45]: I think there's some value on diff. I was considering this being a message for agents so being explicit with what they have to do may be better as these types of users may not have any familiarity with RC.

So giving them clear feedback may help them to fix "faster"

Comment (ggazzo) [reply apps/meteor/client/views/room/composer/ComposerOmnichannel/ComposerOmnichannelInquiry.tsx:45]: If this is the case, considering lay users the tool, the message should not be: "you can't take it because you are offline" instead of "you are offline" ??

Comment (KevLehman) [reply apps/meteor/client/views/room/composer/ComposerOmnichannel/ComposerOmnichannelInquiry.tsx:45]: I'm totally fine with that!

@tiagoevanp hehe

File: packages/i18n/src/locales/en.i18n.json [hunk 24]
```diff
@@ -6419,5 +6419,6 @@
   "Close_gallery": "Close gallery",
   "Next_image": "Next Image",
   "Previous_image": "Previous image",
-  "Image_gallery": "Image gallery"
+  "Image_gallery": "Image gallery",
+  "Youre_unavailable": "You’re unavailable"
```
Comment (KevLehman) [inline packages/i18n/src/locales/en.i18n.json:6423]:
```suggestion
  "You_cant_take_chats_unavailable": "You cannot take new conversations because you're unavailable",
  "You_cant_take_chats_offline": "You cannot take new conversations because you're offline"
```

File: apps/meteor/client/views/room/composer/ComposerOmnichannel/ComposerOmnichannelInquiry.tsx [hunk 25]
```diff
@@ -32,11 +36,24 @@ export const ComposerOmnichannelInquiry = (): ReactElement => {
 		}
 	};
 
-	const t = useTranslation();
+	const title = useMemo(() => {
+		if (user?.status === 'offline') {
+			return t('meteor_status_offline');
```
Comment (KevLehman) [inline apps/meteor/client/views/room/composer/ComposerOmnichannel/ComposerOmnichannelInquiry.tsx:41]:
```suggestion
			return t('You_cant_take_chats_offline');
```

File: apps/meteor/client/views/room/composer/ComposerOmnichannel/ComposerOmnichannelInquiry.tsx (hunk 23 above)
Comment (KevLehman) [inline apps/meteor/client/views/room/composer/ComposerOmnichannel/ComposerOmnichannelInquiry.tsx:45]:
```suggestion
			return t('You_cant_take_chats_unavailable');
```

**Review by KevLehman — CHANGES_REQUESTED** [review]

**Review by KevLehman — DISMISSED** [review]

**Review by MarcosSpessatto — DISMISSED** [review]

**Review by KevLehman — APPROVED** [review]

File: apps/meteor/tests/e2e/page-objects/fragments/home-sidenav.ts [hunk 26]
```diff
@@ -48,6 +48,10 @@ export class HomeSidenav {
 	get sidebarToolbar(): Locator {
 		return this.page.getByRole('toolbar', { name: 'Sidebar actions' });
 	}
+  
+	// Note: this is different from openChat because queued chats are not searchable
+	getQueuedChat(name: string): Locator {
+		return this.page.locator('[data-qa="sidebar-item-title"]', { hasText: name }).first();
```
Comment (KevLehman) [inline apps/meteor/tests/e2e/page-objects/fragments/home-sidenav.ts:54]: image

**Review by MarcosSpessatto — APPROVED** [review]

**Review by ggazzo — APPROVED** [review]
